src/common/catalog/catalog-entity.ts

```diff
@@ -47,7 +48,13 @@ export interface CatalogCategorySpec {
   };
 }
 
-export abstract class CatalogCategory extends EventEmitter {
+export interface CatalogCategoryEvents {
+  onLoad: () => void;
+  onCatalogAddMenu: (context: CatalogEntityAddMenuContext) => void;
+  onContextMenuOpen: (entity: CatalogEntity, context: CatalogEntityContextMenuContext) => void;
+}
+
+export abstract class CatalogCategory extends (EventEmitter as new () => TypedEmitter<CatalogCategoryEvents>) {
```

Comment:
I don't think that that an `EventEmitter` should "await" the events.

From that point of view, `onLoad` should just be emitted and should come with an `offLoad` as well. And both of those should just be "emitted".

Reply:
I agree, Lens shouldn't wait for each extension onLoad event handler to be completed, as that could cause "slow" opening of the Catalog (if I understand this correctly). Instead we should display what we have, and then each extension can asynchronously add new/remove old catalog entities in the event handler.

Reply:
I agree, it does not make sense to wait `onLoad` (I'll remove it). But does it make sense to wait `onContextMenuOpen`?

Reply:
Good question.. I can see benefits/drawbacks for both. If we wait, then the UI can be sluggish if there is at least one extension doing some API calls to add something to the context menu. On the other hand, if we don't wait, then the context menu will open, and a moment later new stuff will be added.

Maybe ideal would be to add a "race", e.g. wait maximum of 50-100ms for the extensions to handle the context menu open event?

Reply:
> I agree, it does not make sense to wait `onLoad` (I'll remove it). But does it make sense to wait `onContextMenuOpen`?

See https://github.com/lensapp/lens/pull/2838 for how I think this should be changed (with regards to menu handlers

Reply:
Refactored code to use just `emit`.